File: CS/Dashboard_EditData/DashboardViewerForm.cs
using System.Collections.Generic;
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars;

namespace Dashboard_EditData {
    public partial class DashboardViewerForm : RibbonForm {
        public DashboardViewerForm() {
            InitializeComponent();
        }

        private void dashboardViewer1_DashboardItemClick(object sender, 
            DashboardItemMouseActionEventArgs e) {
            if (e.DashboardItemName == "gridDashboardItem1" & e.GetUnderlyingData() != null) {
                ShowUnderlyingData(e.GetUnderlyingData());
            }
        }

        private void barButtonItem1_ItemClick(object sender, ItemClickEventArgs e) {
            ShowUnderlyingData(dashboardViewer1.GetUnderlyingData("gridDashboardItem1"));
        }

        private void barButtonItem2_ItemClick(object sender, ItemClickEventArgs e) {
            dashboardViewer1.ReloadData();
        }

        private void ShowUnderlyingData(DashboardUnderlyingDataSet unData) {
            UnderlyingDataForm form = new UnderlyingDataForm();
            string keyColumnName = "OrderID";

            List<object> filterList = new List<object>();
            foreach (DashboardUnderlyingDataRow row in unData) {
                filterList.Add(row[keyColumnName]);
            }
            form.SetFilter(keyColumnName, filterList);

            form.ShowDialog();
            form.Dispose();
        }
    }
}

File: VB/Dashboard_EditData/DashboardViewerForm.vb
Imports System.Collections.Generic
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars

Namespace Dashboard_EditData
    Partial Public Class DashboardViewerForm
        Inherits RibbonForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub dashboardViewer1_DashboardItemClick(ByVal sender As Object, _
                                                        ByVal e As DashboardItemMouseActionEventArgs) _
                                                    Handles dashboardViewer1.DashboardItemClick
            If e.DashboardItemName = "gridDashboardItem1" And e.GetUnderlyingData() IsNot Nothing Then
                ShowUnderlyingData(e.GetUnderlyingData())
            End If
        End Sub

        Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) _
            Handles barButtonItem1.ItemClick
            ShowUnderlyingData(dashboardViewer1.GetUnderlyingData("gridDashboardItem1"))
        End Sub

        Private Sub barButtonItem2_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) _
            Handles barButtonItem2.ItemClick
            dashboardViewer1.ReloadData()
        End Sub

        Private Sub ShowUnderlyingData(ByVal unData As DashboardUnderlyingDataSet)
            Dim form As New UnderlyingDataForm()
            Dim keyColumnName As String = "OrderID"

            Dim filterList As New List(Of Object)()
            For Each row As DashboardUnderlyingDataRow In unData
                filterList.Add(row(keyColumnName))
            Next row
            form.SetFilter(keyColumnName, filterList)

            form.ShowDialog()
            form.Dispose()
        End Sub
    End Class
End Namespace
